torch/distributed/_shard/checkpoint/utils.py

```diff
@@ -0,0 +1,236 @@
+from typing import List, Callable, Tuple, Optional, Union, TypeVar, cast
+import torch.distributed as dist
+from .api import CheckpointException
+import torch
+
+def tensor_narrow_n(tensor: torch.Tensor, offsets: Tuple[int, ...], lengths: Tuple[int, ...]) -> torch.Tensor:
```

Comment:
Could we merge these two util together as they are basically the same? https://github.com/pytorch/pytorch/blob/4bf2f140b369ea85eda4a4d8bdef0263e4b46749/torch/distributed/_shard/_utils.py#L4

Reply:
good point. done.